lib/xgu/xgu.h

```diff
@@ -712,6 +779,99 @@ uint32_t* xgu_set_light_local_attenuation(uint32_t* p, unsigned int light_index,
     return push_floats(p, v.f, 3);
 }
 
+/* ==== Texture stuff ==== */
+
+inline
+uint32_t* xgu_set_texture_offset(uint32_t* p, uint32_t offset) {
+    return push_command_parameter(p, NV097_SET_TEXTURE_OFFSET, offset);
+}
+
+inline
+uint32_t* xgu_set_texture_format(uint32_t* p, uint8_t context_dma, bool cubemap_enable,
```

Comment:
`context_dma` should maybe be an enum, or maybe a `bool`. I'm not sure yet.

As far as I remember, those are typically "A" or "B" which picks a DMA channel.
The vertex attrib stuff should have something like this too I think? How do we handle it there?